src/service/api.js

```diff
@@ -0,0 +1,55 @@
+export const API_END_POINT = "https://kdt-frontend.programmers.co.kr/documents";
+
+export const request = async (url, options = {}) => {
+    try {
+        const res = await fetch(`${API_END_POINT}${url}`, {
+            ...options,
+            headers: {
+                "Content-Type": "application/json",
+                "x-username": "ShinJongUk",
+            },
+        });
+        if (res.ok) {
+            return await res.json();
+        }
+        throw new Error("API 처리중 문제 발생");
+    } catch (error) {
+        console.error(error.message);
+    }
+};
+
+export const getRootDocument = async (url = "") => {
+    const data = await request(url);
+
+    return data;
+};
```

Comment:
```suggestion
export const getRootDocument = async (url = "") => {
    return request(url);
};
```

이 함수는 내부에서 별도의 예외처리가 없어 위와 같은 형태와 동치라고 볼 수 있는데요.
이런 경우 `getRootDocument(url) = request(url)` 또한 동치라 결과적으로는 getRootDocument 함수의 의미가 없다는 결론을 도출할 수 있습니다!

이 경우 2가지 옵션을 선택할 수 있을 것 같슴다!
1. getRootDocument 함수가 외부에서 url을 받지 않고 `request('/documents')`와 같이 API Endpoint까지 합쳐진 결과를 반환한다.
2. 제거한다.

Reply:
커피챗 때 배운 추상화에 대해 잘 생각해보고 잘 고민해서 해보겠습니다 ㅎㅎ 😊👍
좋은 의견 감사합니다!!!! 👍